packages/meditrak-app-server/src/__tests__/__integration__/sync/CountChangesRoute.test.ts

```diff
@@ -0,0 +1,299 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { constructAccessToken } from '@tupaia/auth';
+import { clearTestData, getTestDatabase, getTestModels, upsertDummyRecord } from '@tupaia/database';
+import { TestableServer } from '@tupaia/server-boilerplate';
+import { oneSecondSleep, randomIntBetween, createBearerHeader } from '@tupaia/utils';
+import { ServerChangeEnqueuer } from '../../../sync';
+import { MeditrakAppServerModelRegistry } from '../../../types';
+import { TestModelRegistry } from '../../types';
+import { setupTestApp, setupTestUser } from '../../utilities';
+import { CAT_USER_SESSION } from '../fixtures';
+
+const upsertQuestion = async (models: TestModelRegistry) => {
+  const dataElement = await upsertDummyRecord(models.dataElement, {
+    service_type: 'tupaia',
+  });
+  return upsertDummyRecord(models.question, {
+    code: dataElement.code,
+    data_element_id: dataElement.id,
+  });
+};
+
+describe('changes/count', () => {
```

Comment:
These tests are fantastic!! Thanks to @edmofro who wrote them in [central-server](https://github.com/beyondessential/tupaia/blob/00eb28394161dee852f873a3a452e9283d0f2877/packages/central-server/src/tests/apiV2/getChanges.test.js) I was basically able to copy them over and they worked immediately. Helped me pick out some bugs and gave me a lot more confidence in the code 👍

Reply:
Great! Love an endpoint level test, so robust to internal refactors 😀